app/models/device.rb

```diff
@@ -55,7 +67,7 @@ def resolve_privilege(unresolved_checkins, permissible)
     return Checkin.none if privilege_for(permissible) == 'disallowed'
     return unresolved_checkins if unresolved_checkins.empty?
     if privilege_for(permissible) == 'last_only'
-      Checkin.where(id: unresolved_checkins.first.id)
+      unresolved_checkins.limit(1)
```

Comment:
Changed to limit here so permitted_history_for always returns the same class of object, Checkin.where returned something slightly different than all the other returns in this method.